go/vt/vtgate/vcursor_impl.go

```diff
@@ -104,7 +107,6 @@ type vcursorImpl struct {
 	topoServer     *topo.Server
 	logStats       *logstats.LogStats
 	collation      collations.ID
-	collationEnv   *collations.Environment
```

Comment:
Removed it from here since I realized with the parser, we can always get it from the executor already anyway. Keeping it consistent between the parser and collation env this way.